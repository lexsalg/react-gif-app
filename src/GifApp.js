import { useState } from 'react';
import { CategoryAdd } from './components/CategoryAdd';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({ defaultCategories = [] }) => {
  // const [categories, setCategories] = useState(['One Punch']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif App</h2>
      <CategoryAdd setCategories={setCategories} />
      <hr />
      <ol>
        {
          categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))
        }
      </ol>
    </>
  );
};
