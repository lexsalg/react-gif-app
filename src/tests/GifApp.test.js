import { shallow } from 'enzyme'
import { GifApp } from './../GifApp';


describe('Pruebas en <GifApp />', () => {

    test('should be displayed correctly', () => {

        const wrapper = shallow(<GifApp />);
        expect(wrapper).toMatchSnapshot();

    })


    test('debe de mostrar una lista de categorÃ­as', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })


})
