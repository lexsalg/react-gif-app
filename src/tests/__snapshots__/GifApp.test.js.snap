// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifApp /> should be displayed correctly 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
